### ISSUE TITLE
Bug when calling a function component with `[<ReactComponent>]` directly from another function component with a `[ReactComponent]` attribute

### ISSUE THREAD

**Issue by Lenne231** [Zaid-Ajaj/Feliz#290]
Hey Zaid,

thanks for your great work. Working with the new `[<ReactComponent>]` is fun to work with!

I found a bug when calling a function component with `[<ReactComponent>]` directly from another function component with a `[ReactComponent]` attribute.

# Exptected behavior

```fsharp
[<ReactComponent>]
let MyComponent2 (value: string, onChange: string -> Unit) =
  MyComponent ("Name", value, onChange)
```
compiles to the following javascript

```js
export function MyComponent2(myComponent2InputProps) {
    const onChange = myComponent2InputProps.onChange;
    const value = myComponent2InputProps.value;
    return createElement(MyComponent, {
        label: "Name",
        value: value,
        onChange: onChange,
    });
}
```
# Actual Behavior

the same F# code compiles to

```js
export function MyComponent2(value, onChange) {
    return createElement(MyComponent, {
        label: "Name",
        value: value,
        onChange: onChange,
    });
}
```

# The whole example

I think the problem is that the only thing i'm calling in `MyComponent2` is `MyComponent`.

```fsharp
module Test

open Feliz

// ok
[<ReactComponent>]
let MyComponent (label: string, value: string, onChange: string -> Unit) =
  Html.textf "Value: %s" value

// ok
[<ReactComponent>]
let MyComponent4 (value: string, onChange: string -> Unit) =
  React.fragment [
    MyComponent ("Name", value, onChange)
  ]

// not ok
[<ReactComponent>]
let MyComponent2 (value: string, onChange: string -> Unit) =
  MyComponent ("Name", value, onChange)

let myComponent2 = MyComponent2("", fun _ -> ())
```

```js
import { printf } from "./.fable/fable-library.3.0.0/String.js";
import { createElement } from "react";
import * as react from "react";

export function MyComponent(myComponentInputProps) {
    const onChange = myComponentInputProps.onChange;
    const value = myComponentInputProps.value;
    const label = myComponentInputProps.label;
    return printf("Value: %s").cont((arg00) => arg00)(value);
}

export function MyComponent4(myComponent4InputProps) {
    const onChange = myComponent4InputProps.onChange;
    const value = myComponent4InputProps.value;
    return react.createElement(react.Fragment, {}, createElement(MyComponent, {
        label: "Name",
        value: value,
        onChange: onChange,
    }));
}

export function MyComponent2(value, onChange) {
    return createElement(MyComponent, {
        label: "Name",
        value: value,
        onChange: onChange,
    });
}

export const myComponent2 = createElement(MyComponent2, {
    value: "",
    onChange: (_arg1) => {
    },
});
```

**Comment by Zaid-Ajaj** [comment]
Hi there @Lenne231, thanks a lot for the detailed issue, I will have a look

**Comment by Zaid-Ajaj** [comment]
Fixed as of Feliz v1.24 🚀 can you please give it a try to see if issue is resolved? 

Re-open the issue if the problem persists 😉

**Comment by Lenne231** [comment]
Thanks for the fix, works as expected now :smiley:
